src/Microsoft.IdentityModel.JsonWebTokens/JsonWebTokenHandler.cs

```diff
@@ -150,11 +150,21 @@ public virtual string CreateToken(SecurityTokenDescriptor tokenDescriptor)
             if (tokenDescriptor == null)
                 throw LogHelper.LogArgumentNullException(nameof(tokenDescriptor));
 
-            if (tokenDescriptor.Claims == null || !tokenDescriptor.Claims.Any())
-                LogHelper.LogWarning(LogMessages.IDX14114);
+            if ((tokenDescriptor.Subject == null || !tokenDescriptor.Subject.Claims.Any()) 
+                && (tokenDescriptor.Claims == null || !tokenDescriptor.Claims.Any()))
+                LogHelper.LogWarning(LogMessages.IDX14114, nameof(SecurityTokenDescriptor), nameof(SecurityTokenDescriptor.Subject), nameof(SecurityTokenDescriptor.Claims));
+            
+            JObject payload;
+            if (tokenDescriptor.Subject != null)
+                payload = JObject.FromObject(JwtTokenUtilities.CreateDictionaryFromClaims(tokenDescriptor.Subject.Claims));
+            else
+                payload = new JObject();
 
-            // JObject needs to be upcast to an IDictionary<string, JToken> so that we can access the ContainsKey() and Any() methods
-            IDictionary<string, JToken> payload = tokenDescriptor.Claims == null ? new JObject() : JObject.FromObject(tokenDescriptor.Claims);
+            // If a key is present in both tokenDescriptor.Subject.Claims and tokenDescriptor.Claims, but the key is not the same case in both (e.g. "key" and "KEY"),
```

Comment:
I believe that this comment is no longer true.

Reply:
You're right! I fixed the comment so it now describes the current logic.